### PR TITLE
test: [M3-8316,M3-8317] - Tag cypress tests by adding the "method:e2e" and "purpose:dcTesting"

### PR DESCRIPTION
**Pull request by AzureLatte**
## Description 📝
Tag tests by adding the "method:e2e" tag and "purpose:dcTesting".

## Changes  🔄
- Use cy.tag() to add the "method:e2e" tag to all of our end-to-end Cypress tests.
- Use cy.tag() to add the "purpose:dcTesting" tag to all of our suitable Cypress tests.
- Add new intercept utils in linodes.ts and replace them in end-to-end tests.

## How to test 🧪
CI tests all pass. Or running tests locally using:
```
yarn cy:run
```

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [ ] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [ ] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/cypress/e2e/core/linodes/linode-storage.spec.ts [hunk 1]
```diff
@@ -105,10 +112,7 @@ describe('linode storage tab', () => {
   it('try to delete in use disk', () => {
     const diskName = 'Debian 11 Disk';
     cy.defer(() => createTestLinode({ booted: true })).then((linode) => {
-      cy.intercept(
-        'DELETE',
-        apiMatcher(`linode/instances/${linode.id}/disks/*`)
-      ).as('deleteDisk');
+      interceptDeleteDisks(linode.id).as('deleteDisk');
```
Comment (jdamore-linode) [inline packages/manager/cypress/e2e/core/linodes/linode-storage.spec.ts:115]: Thanks for catching and cleaning all of these intercepts up! 🧼

File: packages/manager/cypress/e2e/core/nodebalancers/smoke-create-nodebal.spec.ts [hunk 2]
```diff
@@ -78,6 +78,10 @@ const createNodeBalancerWithUI = (
 };
 
 authenticate();
+beforeEach(() => {
+  cy.tag('method:e2e');
+  cy.tag('purpose:dcTesting');
```
Comment (jdamore-linode) [inline packages/manager/cypress/e2e/core/nodebalancers/smoke-create-nodebal.spec.ts:83]: `cy.tag()` will override previous calls to `cy.tag()`, so these tests only end up with the `purpose:dcTesting` tag.

You can set both tags at once with `cy.tag('method:e2e', 'purpose:dcTesting')` which is probably the right call in this case, but we also have `cy.addTag()` to add tags without overriding existing tags.

I think there are a couple places we'll want to make this fix!

**Review by jdamore-linode — APPROVED** [review]
This is so cool, thanks @AzureLatte!

I'm almost looking forward to having to run our next DC tests* -- it's going to be so nice to call `CY_TEST_TAGS='purpose:dcTesting' yarn cy:run` and not have to wait for every test to run.

I did notice a couple spots where we call `cy.tag()` multiple times and made a suggestion around that, but otherwise this is good to go! Thanks again for the extra clean up!!

**Review by abailly-akamai — APPROVED** [review]
Thanks! Approving pending fix suggested

**Pull request merged**